src/session_impl.cpp

```diff
@@ -1697,7 +1708,6 @@ namespace aux {
 #ifndef TORRENT_NO_DEPRECATE
 	void session_impl::set_settings(libtorrent::session_settings const& s)
 	{
-		INVARIANT_CHECK;
```

Comment:
We could make the impl method check free as you did suggest.
Then we could call apply_settings_pack_impl method instead of apply_settings_pack to keep the stille "invarient check marko at set method start)